LeaderboardBackend/Services/Impl/AccountRecoveryService.cs

```diff
@@ -55,8 +59,9 @@ public async Task<CreateRecoveryResult> CreateRecoveryAndSendEmail(User user)
                 GenerateAccountRecoveryEmailBody(user, recovery)
             );
         }
-        catch
+        catch (Exception e)
         {
+            _logger.LogError("Recovery email failed to send: {err}", e.Message);
```

Comment:
I would be wary of logging the contents of the error message because it could contain the recipient email address or the contents of the email. Instead, consider logging the user ID and optionally the username so we know who was trying to recover their account.

Reply:
While that's true, I don't think just knowing who it is would help diagnosing much. We need some way to fish the problem out better. The underlying service call can fail for a lot of reasons, including failure to connect to the SMTP server, or to log into it.

I'll change to logging the exception type, user ID, and name